aws/s3.tf

```diff
@@ -12,7 +12,18 @@ resource "aws_s3_bucket" "richie_media" {
   }
 
   versioning {
-    enabled = true
+    enabled = var.media_expiration > 0 ? false : true
+  }
+
+  dynamic "lifecycle_rule" {
+    for_each = var.media_expiration > 0 ? [1] : []
```

Comment:
I don't see where you are using it. Is it a hack to not declare this block if `media_expiration` equal `0` ?

Reply:
it's not really a hack but the best way I found to do a conditional blocks in Terraform.
You can consider it as a "if"...